lib/resque/job.rb

```diff
@@ -204,6 +219,14 @@ def perform
       end
     end
 
+    def call_around_hook(job, hook:, args:, &block)
+      if hook.to_s.end_with?('_with_job')
```

Comment:
NOTE: Yes, this is nasty - but I couldn't find a better way to make this determination. Some hooks use `*args`, but that's not a reliable indication. I also considered introspecting for something like a required keyword arg `job:`, but those don't appear to be exposed through the `Method` object.

Reply:
TIL `Method#parameters`:
```
irb(main):001:0> class Foo
irb(main):002:1>   def bar(job:, &block)
irb(main):003:2>     puts job.inspect
irb(main):004:2>   end
irb(main):005:1> end
=> :bar
irb(main):007:0> Foo.new.method(:bar).parameters
=> [[:keyreq, :job], [:block, :block]]
```
So the check would be `job.method(hook).parameters.include?([:keyreq, :job])` 🤔